tests/pytests/functional/modules/test_win_useradd.py

```diff
@@ -0,0 +1,341 @@
+
+pytestmark = [
+    pytest.mark.destructive_test,
+    pytest.mark.skip_unless_on_windows,
+    pytest.mark.windows_whitelisted,
+]
+
+
+@pytest.fixture(scope="module")
+def user(modules):
+    return modules.user
+
+
+@pytest.fixture
+def username_str(user):
+    _username = random_string("test-account-", uppercase=False)
+    try:
+        yield _username
+    finally:
+        try:
+            user.delete(_username, purge=True, force=True)
+        except Exception:  # pylint: disable=broad-except
+            # The point here is just system cleanup. It can fail if no account was created
+            pass
+
+
+@pytest.fixture
+def username_int(user):
+    _username = random_string("", uppercase=False, lowercase=False, digits=True)
+    try:
+        yield _username
+    finally:
+        try:
+            user.delete(_username, purge=True, force=True)
+        except Exception:  # pylint: disable=broad-except
+            # The point here is just system cleanup. It can fail if no account was created
+            pass
+
+
+@pytest.fixture
+def account_str(user, username_str):
+    with pytest.helpers.create_account(username=username_str) as account:
+        user.addgroup(account.username, "Users")
+        yield account
+
+
+@pytest.fixture
+def account_int(user, username_int):
+    with pytest.helpers.create_account(username=username_int) as account:
+        user.addgroup(account.username, "Users")
+        yield account
+
+
+def test_add_str(user, username_str):
+    ret = user.add(name=username_str)
+    assert ret is True
+    assert username_str in user.list_users()
+
+
+def test_add_int(user, username_int):
+    ret = user.add(name=username_int)
+    assert ret is True
+    assert username_int in user.list_users()
+
+
+def test_addgroup_str(user, account_str):
+    ret = user.addgroup(account_str.username, "Backup Operators")
+    assert ret is True
+    ret = user.info(account_str.username)
+    assert "Backup Operators" in ret["groups"]
+
+
+def test_addgroup_int(user, account_int):
+    ret = user.addgroup(account_int.username, "Backup Operators")
+    assert ret is True
+    ret = user.info(account_int.username)
+    assert "Backup Operators" in ret["groups"]
+
+
+def test_chfullname_str(user, account_str):
+    ret = user.chfullname(account_str.username, "New Full Name")
+    assert ret is True
+    ret = user.info(account_str.username)
+    assert ret["fullname"] == "New Full Name"
+
+
+def test_chfullname_int(user, account_int):
+    ret = user.chfullname(account_int.username, "New Full Name")
+    assert ret is True
+    ret = user.info(account_int.username)
+    assert ret["fullname"] == "New Full Name"
+
+
+def test_chgroups_single_str(user, account_str):
+    groups = ["Backup Operators"]
+    ret = user.chgroups(account_str.username, groups=groups)
+    assert ret is True
+    ret = user.info(account_str.username)
+    groups.append("Users")
+    assert ret["groups"].sort() == groups.sort()
```

Comment:
```suggestion
    assert sorted(ret["groups"]) == sorted(groups)
```

`.sort()` does not return anything, while sorted does.


```python
>>> a = [3, 1, 0]
>>> b = [4, 1, 2]
>>> a.sort()
>>> sorted(b)
[1, 2, 4]
```

If you really wanted to use `.sort()`
```python
    ret["groups"].sort()
    groups.sort()
    assert ret["groups"] == groups
```

Reply:
No wonder they all passed...

Reply:
Changed to use sorted instead of sort